### PR TITLE
Added testing for RecordStoreQueryPlugin

### PR DESCRIPTION
**Pull request by arcturus2**
### Description of the Change

* Reviewed and updated unit testing for the RecordStooreQueryPlugin
* Minor updates to the class to resolve some SonarQube warnings

### Alternate Designs

Testing could have been added for the `read`, `query` and `edit` methods separately but there was a lot of dependencies and the same lines of code being run multiple times I felt that putting them into 1 test was better. For instance, you can't run the `query` method without running `read` because `read` is needed to initialise some of the class.

I tried to use Mockito for the 2 `RecordStoreValidator` instances I had to make but I was not able to figure out how to create 2 different types of classes that extend `RecordStoreValidator`. This was required because the class is used to determine what validators are removed.

### Applicable Issues
#413

### CODE REVIEW

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreQueryPluginNGTest.java [hunk 1]
```diff
@@ -65,112 +72,189 @@ public void tearDownMethod() throws Exception {
...
+	final RecordStoreQueryPlugin instance = new RecordStoreQueryPluginMockImpl();
+	final GraphReadMethods graph = new StoreGraph();
+	final PluginInteraction interaction = null;
+	final PluginParameters parameters = null;
+	try {
+	    instance.read(graph, interaction, parameters);
+	    instance.query(interaction, parameters);
+	} catch (InterruptedException | PluginException ex) {
+	    fail(ex.getLocalizedMessage());
+	}
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreQueryPluginNGTest.java:86]: I think throwing the exception is preferred here

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreQueryPluginNGTest.java [hunk 2]
```diff
@@ -65,112 +72,189 @@ public void tearDownMethod() throws Exception {
...
     public void testAddValidator() {
-        final RecordStoreValidator validator = new RecordStoreValidator1MockImpl();
-        final RecordStoreQueryPlugin instance = new RecordStoreQueryPluginMockImpl();
+	System.out.println("addValidator");
+
+	final RecordStoreValidator validator = new RecordStoreValidator1MockImpl();
+	final RecordStoreQueryPlugin instance = new RecordStoreQueryPluginMockImpl();
 
-        assertEquals(instance.getValidators().size(), 0);
+	assertEquals(instance.getValidators().size(), 0);
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreQueryPluginNGTest.java:181]: could also do `assertTrue(instance.getValidators().isEmpty())`

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreQueryPluginNGTest.java [hunk 3]
```diff
@@ -65,112 +72,189 @@ public void tearDownMethod() throws Exception {
...
-        final RecordStoreValidator validator2 = new RecordStoreValidator2MockImpl();
-        final RecordStoreQueryPlugin instance = new RecordStoreQueryPluginMockImpl();
+	System.out.println("removeValidator");
+
+	final RecordStoreValidator validator1 = new RecordStoreValidator1MockImpl();
+	final RecordStoreValidator validator2 = new RecordStoreValidator2MockImpl();
+	final RecordStoreQueryPlugin instance = new RecordStoreQueryPluginMockImpl();
 
-        assertEquals(instance.getValidators().size(), 0);
+	assertEquals(instance.getValidators().size(), 0);
```
Comment (antares1470) [inline CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreQueryPluginNGTest.java:199]: similar here

**Review by antares1470 — CHANGES_REQUESTED** [review]

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreQueryPluginNGTest.java (hunk 1 above)
Comment (arcturus2) [reply CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreQueryPluginNGTest.java]: Agreed and updated.

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreQueryPluginNGTest.java (hunk 2 above)
Comment (arcturus2) [reply CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreQueryPluginNGTest.java]: Updated thanks

File: CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreQueryPluginNGTest.java (hunk 3 above)
Comment (arcturus2) [reply CoreDataAccessView/test/unit/src/au/gov/asd/tac/constellation/views/dataaccess/templates/RecordStoreQueryPluginNGTest.java]: Updated thanks

**Review by antares1470 — APPROVED** [review]

**Pull request merged**